### PR TITLE
Custom button live event

### PR DESCRIPTION
**Pull request by l0ud0gg**
bring custom owned button component into core template. the idea being that if a client needs to use this feature, all we need to do is ensure they are on a recent version of core template, then enable custom fields / create the custom fields. once thats configured, they can populate the link and label on the button for the films they want to add the button to.

### CODE REVIEW

File: site/static/js/custom-owned-button.component.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    }
+  };
+
+  shoppingSessionCompleted(session) {
+    if (session.data.slug != this.props.slug) { return; }
+    this.updateState();
+  }
+
+  loadAvailability(slug){
+    return this.app.itemLimitService.get(slug).then(limit => {
```
Comment (PeculiarGoat) [inline site/static/js/custom-owned-button.component.js:38]: Sorry just noticed this item limit call... it can be removed!

**Review by PeculiarGoat — CHANGES_REQUESTED** [review]
Looks good apart from that one thing which is my bad :p

**Review by PeculiarGoat — APPROVED** [review]

**Review by stajs — COMMENTED** [review]
Sorry, I thought I had saved the review last time - not sure what happened.

File: site/static/js/custom-owned-button.component.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  openLink(event, url) {
+    event.preventDefault();
+    if (url.length == 0) return;
+    window.open(url);
+  }
+
+  render(props, state) {
+    if (!state.loaded || !state.isOwned || props.url == '' || props.label == '') return;
+    return state.loaded && state.isOwned && (
```
[the remark this answers is not in the document]
Comment (l0ud0gg) [reply site/static/js/custom-owned-button.component.js]: commited your suggestion, churrrr

**Pull request merged**